tests/library/precompiled-dxil.slang

```diff
@@ -7,9 +7,9 @@
 // The test passes if there is no errror thrown.
```

Comment:
Looking at the slang-test output in CI, there seems to be a hazard in multi-stage dependent tests like this:

assed test: 'tests/library/export-test.slang' 
passed test: 'tests/library/precompiled-dxil.slang' 
passed test: 'tests/library/library-test.slang' 
passed test: 'tests/library/library-test.slang.1' 
passed test: 'tests/library/export-test.slang.1' 
passed test: 'tests/library/precompiled-dxil.slang.1' 
passed test: 'tests/metal/atomic-texture-buffer.slang' 
passed test: 'tests/library/precompiled-dxil.slang.2' 

Looks like the precompiled-dxil tests are interleaved with other library tests, and since they even use the same export-library shader source, it seems like a genuine risk to look out for.  It so happens that export-test.slang doesn't create an export-library.slang-module to conflict with the export-library.slang-module my test creates, but let me know if I should preemptively fix this, maybe by making my artifact file names more unique.

Reply:
There are multiple threads running in parallel, causing the interleaving of output. However each file is processed in a single thread so it should be fine.

Reply:
I'm unsure. The test files, like "tests/library/library-test.slang" or "tests/library/precompiled-dxil.slang" each contain separate TESTs, and there's an assumption in each file that the output from one TEST is available to the next TEST in that file. But if two different test files are using the same output file names, like "export-library.slang-module", isn't there a hazard that the module file on disk is modified by the other test?  IIUC, the only way to resolve this is to make sure there aren't any output filenames shared between two different test files.  I believe that's true now, but it seems risky to have different tests running in parallel and dumping intermediate files to disk which might have filename collisions by accident.

Reply:
It is the responsibility of each test to make sure they use a distinct name for its intermediate outputs.  This means that if another test is generating export-library.slang-module, then this test shouldn't be generating the samely named file.

Reply:
Currently it's fine and there are no clashes. I was just pointing out this issue in case there was a solution to it that I didn't know about, or if there was a naming requirement that I wasn't following, e.g. prefixing output files with the name of the test.